heroku_connect/models/base.py

```diff
@@ -0,0 +1,223 @@
+from django.core import checks
+from django.db import models
+
+from . import fields
+from ..conf import settings
+
+__all__ = ('HerokuConnectModel',)
+
+
+class HerokuConnectModelBase(models.base.ModelBase):
+    """
+    Custom model base class to set ``Meta`` attributes.
+
+    Sets :attr:`Meta.managed<django.db.models.Options.managed>` to ``False``
+    and sets a default value for :attr:`Meta.db_table<django.db.models.Options.db_table>`.
+    """
+
+    def __new__(mcs, name, bases, attrs):
+        super_new = super(HerokuConnectModelBase, mcs).__new__
+        # We need t create a temporary fake model class to access
+        # the original Meta class since ModelBase will remove it.
+        fake_class = type.__new__(mcs, name, bases, attrs)
+        org_meta_attr = getattr(fake_class, 'Meta', None)
+        new_class = super_new(mcs, name, bases, attrs)
+        if org_meta_attr is None or not hasattr(org_meta_attr, 'db_table'):
+            new_class._meta.db_table = '"{schema}"."{table}"'.format(
+                schema=settings.HEROKU_CONNECT_SCHEMA,
+                table=new_class.sf_object_name.lower(),
+            )
+
+        # User object in Heroku Connect has no is_deleted field.
+        if new_class.sf_object_name == 'User':
+            is_deleted = [x for x in new_class._meta.local_fields if x.name == 'is_deleted'][0]
+            new_class._meta.local_fields.remove(is_deleted)
+
+        new_class._meta.managed = False
+        return new_class
+
+
+class HerokuConnectModel(models.Model, metaclass=HerokuConnectModelBase):
+    """
+    Base model for Heroku Connect enabled ORM models in Django.
+
+    Example::
+
+        from heroku_connect import models as hc_models
+
+
+        class MyCustomObject(hc_models.HerokuConnectModel):
+            sf_object_name = 'My_Custom_Object__c'
+
+            custom_date = hc_models.DateTime(sf_field_name='Custom_Date__c')
+
+    Note:
+
+        Subclasses have :attr:`Meta.managed<django.db.models.Options.managed>` set to ``False``.
+
+        A default value for :attr:`Meta.db_table<django.db.models.Options.db_table>` is set based
+        on :attr:`settings.HEROKU_CONNECT_SCHEMA<.HerokuConnectAppConf.HEROKU_CONNECT_SCHEMA>`
+        and :attr:`.sf_object_name`.
+
+    Warning:
+
+        The Salesforce object `User`_ object ``IsDeleted`` field. Therefore
```

Comment:
something's missing in this sentence (... object **has no** ``IsDeleted`` ... ?)